crates/runtime/src/http/routes.rs

```diff
@@ -242,13 +242,28 @@ pub(crate) fn routes(
     unauthenticated_router
         .merge(authenticated_router)
         .route_layer(middleware::from_fn_with_state(rt.status(), check_shutdown))
+        .route_layer(middleware::from_fn(with_datafusion_context))
         .route_layer(middleware::from_fn(track_metrics))
         .layer(Extension(Arc::clone(&rt.app)))
+        .layer(Extension(Arc::clone(&rt.df)))
```

Comment:
I want to force the code that relies on a `DataFusion` instance to always get it from the request context, so we should add this layer only to the `track_metrics` middleware layer, which will remove it from all other handlers.

Something like:

```rust
let track_metrics_with_df = middleware::from_fn(track_metrics)
        .layer(Extension(Arc::clone(&rt.df)));
..

.route_layer(track_metrics_with_df)
```

Reply:
Otherwise, we're now in a place where we can get it either from the current Extension layer or from the request context.

Reply:
updated